paasta_tools/mesos_maintenance.py

```diff
@@ -106,7 +105,7 @@ def maintenance_api():
     """
     def execute_schedule_api_request(method, endpoint, **kwargs):
         master_api_client = master_api()
-        return master_api_client(method, "/maintenance%s" % endpoint, **kwargs)
+        return master_api_client(method, "/maintenance%s" % endpoint, timeout=(3, 3), **kwargs)
```

Comment:
This is is still a bit too optimistic. Loading the dashboard in my browser I can see that it takes ~2.3 second in the normal case, sometimes it is better, but in loaded times (like yesterday) it is going to be worse.

How about 10s?